src/main/java/io/aiven/kafka/connect/transforms/ConcatFields.java

```diff
@@ -0,0 +1,134 @@
+            }
+            newSchema.field(config.outputFieldName(), Schema.OPTIONAL_STRING_SCHEMA);
+            final Struct newStruct = new Struct(newSchema);
+            struct.schema().fields().forEach(field -> {
+                newStruct.put(field.name(), struct.get(field));
+            });
+            config.fieldNames().forEach(field -> {
+                try {
+                    if (struct.get(field) == null) {
+                        fieldValues.add(config.fieldReplaceMissing());
+                    } else {
+                        fieldValues.add(struct.get(field));
+                    }
+                } catch (final Exception e) {
+                    log.debug("{} is missing, concat will use {}", field, config.fieldReplaceMissing());
+                    fieldValues.add(config.fieldReplaceMissing());
+                }
+            });
+            final StringJoiner outputValue = new StringJoiner(config.delimiter());
+            for (final Object fieldValue : fieldValues) {
+                final String toString = fieldValue.toString();
+                outputValue.add(toString);
+            }
+            newStruct.put(config.outputFieldName(), outputValue.toString());
+            newRecord = createNewRecord(record, newSchema.build(), newStruct);
+        } else if (record.value() instanceof Map) {
+            final Map newValue = (Map<?, ?>) record.value();
+            config.fieldNames().forEach(field -> {
+                if (newValue.get(field) == null) {
+                    fieldValues.add(config.fieldReplaceMissing());
+                } else {
+                    fieldValues.add(newValue.get(field));
+                }
+            });
+            final StringJoiner outputValue = new StringJoiner(config.delimiter());
+            for (final Object fieldValue : fieldValues) {
+                final String toString = fieldValue.toString();
+                outputValue.add(toString);
+            }
+            newValue.put(config.outputFieldName(), outputValue.toString());
+            newRecord = createNewRecord(record, null, newValue);
+        } else {
+            throw new DataException("Value type must be STRUCT or MAP: " + record);
+        }
+
+        return newRecord;
+    }
+
+    private final R createNewRecord(final R record, final Schema newSchema, final Object newValue) {
+        return record.newRecord(
+            record.topic(),
+            record.kafkaPartition(),
+            record.keySchema(),
+            record.key(),
+            newSchema,
+            newValue,
+            record.timestamp(),
+            record.headers()
+        );
+    }
+
+    public static final class Value<R extends ConnectRecord<R>> extends ConcatFields<R> {
+        // There's an implementation only for value, not for key.
+        // We provide $Value class anyway for the consistency sake
+        // and in case we need a $Key version in the future as well.
+    }
```

Comment:
The documentation says both classes are implemented. The implementation (+ tests) is needed.